[DOR-47] Serve IIIF manifest from Rosetta

tm-kn opened the pull request:
Ticket URL: https://national-archives.atlassian.net/browse/DOR-47

## About these changes

- Add an API call to the new IIIF manifest endpoint.
- The prefix varies to the existing API endpoint, so had to add a new environment variable: `_CLIENT_IIIF_MANIFEST_BASE_URL`.
- Add a fetch call at the record detail view to determine if the record has IIIF manifest generated. This is a bit inefficient as we have to make a request even if the record has no IIIF manifest (Raised https://national-archives.atlassian.net/browse/DOR-53 to address it).
- ~Add a new view at `/catalogue/id/<IAID>/iiif-manifest/` to serve the manifests to the IIIF viewer JavaScript front-end.~
- Decided against implementing the proxy view. Sounds like a counterintuitive idea at the moment. The manifest can link out to a lot of different resources that the viewer may want to call itself. Proxying the view, would mean having to proxy all other resources and that would require implementing quite a complicated router. It feels like this complexity shouldn't belong in the Django web application.

## How to check these changes

View record detail page locally, e.g. http://localhost:8000/catalogue/id/C4700551/

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [ ] Waited for all CI jobs to pass before requesting a review
- [ ] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

tm-kn commented on etna/records/views/records.py:
```diff
@@ -124,3 +147,34 @@ def record_detail_view(request, id):
+        return HttpResponse(status=503)
+
+    # Proxy the IIIF manifest as-is from the CIIM (Rosetta) API to the client.
+    return JsonResponse(iiif_manifest.content)
```
**Issue (blocking):** I think there's a big problem with using this proxy. The IIIF manifest contains links to the API that must be resolvable by the end user. I'm not sure how we want to proceed with that.

jamesbiggs replied:
Sorry Tomasz, I can't view the link you shared as it's behind Torchbox's slack. I've not seen the manifests etc so not sure what the real difference is apart from them being separate (for now)

tm-kn replied:
Sorry! It was a thread on our shared channel with Lee. I've got rid of the proxy view for now because it does not seem feasible to implement it now.

tm-kn commented on etna/ciim/client.py:
```diff
@@ -296,6 +300,37 @@ def fetch(
+            Matches a single IAID.
+            Ex: returns match on Information Asset Identifier - iaid.
+        """
+        response = self.make_request(self.get_public_iiif_manifest_url(id=id))
```
**Question:** Is there a need to implement a fake request of some kind for the CI tests to not hit the real server?

jamesbiggs replied:
I think we have some `records.json`s that we use for this - I haven't been too involved in the "search" applications though, so not sure what state we're in at the minute. If you want to make some tests that you feel are appropriate, feel free! 😃

tm-kn replied:
Mhm... I am hesitant to add tests I can't run with confidence because the existing unit tests are not ready for Rosetta. I don't want to silently introduce broken tests. I'll push the testing to #1554.

jamesbiggs reviewed (approved):
LGTM! 🚀

tm-kn commented on etna/records/views/records.py:
```diff
@@ -87,11 +89,38 @@ def record_detail_view(request, id):
+        #       so we know in advance if we need to call the IIIF manifest
+        #       endpoint at all. The raised ticket:
+        #       https://national-archives.atlassian.net/browse/DOR-53
+        records_client.fetch_iiif_manifest(id=record.iaid)
```
**Question:** Should this only run for certain record types? E.g. we might not want to make the request for `ARCHON` and `CREATEORS` record types. They also have different template which means that the viewer is not implemented there.

jamesbiggs replied:
Yes, I think we only want this for `digitised` records which cover all digitised records at TNA. I don't think ARCHON or CREATORS will have any images/digitised stuff. Not sure if it'd be easy to prevent it running on ARCHON etc?

tm-kn replied:
Just added an if-statement, same as for the template selection. An observation: It feels like the view as a function-based view looks quite messy given it does so much right now. There's a lot of noise all in one function. It may be beneficial to implement it as a class-based view or break down the function at some point.

tm-kn commented:
@jamesbiggs Sorry to be asking you to review this multiple times. I made an update to the code:
- Only fetching IIIF manifest for the digitised records.
- Only fetching IIIF manifest for templates that have the IIIF viewer.
- Added tests for the records client in this PR.
- Added additional tests for the record detail view in #1558. Can't do this as part of this as the unit testing for Rosetta is not working so I can't verify my tests are correct.

jamesbiggs reviewed (approved):
All looks good, cheers Tomasz! 😄
